client/src/app/components/tool-container/tool-header/tool-header.component.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    Component,
+    EventEmitter,
+    Input,
+    Output
+} from "@angular/core";
+import {NgSelectOption} from "@angular/common";
+import {ViewMode} from "../tool-container.component";
+
+require("./tool-header.component.scss");
+
+@Component({
+    selector: "tool-header",
+    directives: [NgSelectOption],
+    template: `
+        <header class="toolHeader">
+                <span class="guiJsonButtons">
+                    <button type="button"
+                            class="btn btn-secondary btn-sm selected toolHeaderBtn"
+                            [ngClass]="{selected: viewMode === 'json'}"
+                            (click)="changeViewMode('json')">JSON</button>
+
+                    <button type="button toolHeaderBtn"
```

Comment:
`toolHeaderBtn` is not a valid value for the `type` attribute of a  `<button>` element.